### ISSUE TITLE
:bug: require-hyphen-before-param-description places hyphen in the wrong place on multiline types

### ISSUE THREAD

**Issue by scagood** [gajus/eslint-plugin-jsdoc#1324]
## Code sample -- Before '--fix'
```js
/**
 * @param {(
 *  | string
 *  | number
 * )} input The input value
 */
function test(input) {}
```

## Expected behavior -- After '--fix'
```ts
/**
 * @param {(
 *  | string
 *  | number
 * )} input - The input value
 */
function test(input) {}
```

## Actual behavior -- After '--fix'
```ts
/**
 * @param {(- - - - - - - - - - 
 *  | string
 *  | number
 * )} input The input value
 */
function test(input) {}
```

## ESLint Config

```ts
import jsdoc from 'eslint-plugin-jsdoc';

export default [
  {
    plugins: { jsdoc },
    rules: {
      'jsdoc/require-hyphen-before-param-description': 'warn'
    }
  }
];
```

## Reproduction
To reproduce, click 'Apply Fix' on the `src/example.js` file.

https://eslint-online-playground.netlify.app/#eNpVULtygzAQ/JUbNdiMgR43KfIJ6aIUsjhAjjgpejjOEP49SODMuDlpb1fa25uZd7LBu5isxvrqWcuasuQEJbxY4cQE8yEj+AUfnKLhgShOF3QZHRdQZGOAtxH3203oiIlsOPWRZFCGIKAPh8wfYV7YiaHXikItDfVqqKdsryZrXICr74yE3pkJik1WWR0HRVVmijMnTnjP2g57EXWA99UQ5lQANrFvYd6/Wk4b4aLG1N4QQJHpxuFXVA6r8ceOSNUFe7OivIKqQy+dsilD0ULxLRwV2/MlHWv5OK9prJCfYkhbNLQGyQ6cdXh7RYvUIUmFnrN/b77nTy3OtEjr4Wwf80E+pX5W7t4LW/4AYsCY3A==

**Comment by brettz9** [comment]
Fixed by #1325 . Released in v50.3.2. Thanks for the report!

**Comment by scagood** [comment]
Thanks @brettz9!
